### ISSUE TITLE
Method of quickly disabling GPU partitioning

### ISSUE THREAD

**Issue by kietatran00** [jamesstringerparsec/Easy-GPU-PV#398]
Hello! I found that even when the VM is offline it'll still partition the GPU and end up losing performance when playing other games on the host. I was able to use Remove-VMGpuPartitionAdapter and removed the reserved GPU for the VM and noticed a drastic FPS increase, however I couldn't find a way to easily add the partionedGPU back to the VM. Any advice?

**Comment by iatoba72** [comment]
also interested in this !! an easy way to: when shuting down the vm, being able to use 100% on the host. maybe a rollback script?

**Comment by Stefanuk12** [comment]
I'm not the best at batch scripts, here's what I was able to come up with... Make sure you run as Administrator.
[(de)attach.zip](https://github.com/user-attachments/files/17000073/de.attach.zip)

**Comment by Stefanuk12** [comment]
I Frankenstein-ed this one together... Make sure you run as Administrator.
[(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)

**Comment by jordanfalduto** [comment]
> Hello! I found that even when the VM is offline it'll still partition the GPU and end up losing performance when playing other games on the host. I was able to use Remove-VMGpuPartitionAdapter and removed the reserved GPU for the VM and noticed a drastic FPS increase, however I couldn't find a way to easily add the partionedGPU back to the VM. Any advice?

how would you even be able to confirm that 100% of the GPU wasn't being utilized besides an FPS test?

**Comment by DPoveaa** [comment]
> I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)

And how can i attach again?

**Comment by Stefanuk12** [comment]
> > I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)
> 
> 
> 
> And how can i attach again?

change the mode to `Attach`

**Comment by jordanfalduto** [comment]
> > > I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)
> > 
> > 
> > And how can i attach again?
> 
> change the mode to `Attach`

So the $params.Mode is set to detach if the Mode = "Deattach"
but if $params.Mode is anything else it will Attach according to line 10 through 40?

**Comment by Stefanuk12** [comment]
> > > > I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)
> 
> > > 
> 
> > > 
> 
> > > And how can i attach again?
> 
> > 
> 
> > change the mode to `Attach`
> 
> 
> 
> So the $params.Mode is set to detach if the Mode = "Deattach"
> 
> but if $params.Mode is anything else it will Attach according to line 10 through 40?

Yeah, I say to make it `Attach` for the sake of simplicity but you're totally right. You can set it to anything else.

**Comment by DPoveaa** [comment]
> > > I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)
> > 
> > 
> > And how can i attach again?
> 
> change the mode to `Attach`

Sorry for the stupid question, but how do I confirm that it worked? And whether or not the virtual machine has a video card linked?

**Comment by Stefanuk12** [comment]
> > > > I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)
> 
> > > 
> 
> > > 
> 
> > > And how can i attach again?
> 
> > 
> 
> > change the mode to `Attach`
> 
> 
> 
> Sorry for the stupid question, but how do I confirm that it worked? And whether or not the virtual machine has a video card linked?

If you try to remove the adapter twice, it should error. If it errors, you know that the specified card isn't linked.

**Comment by jordanfalduto** [comment]
I finally got around to testing this and I don't really think this is a thing

1. It's not how hosts work, when a VM is powered off, it consumes no resources

2. Stefanuk12's PS script absolutely does it's intended purpose and rips the GPU from the VM. I detached it and watched it get ripped from the powered on VM in real time. System Information > Components > Display no longer shows the host GPU

3. My performance did not increase or decrease after detaching. Pulling up Task Manager I see my GPU resources are at 100% utilization before and after. My GPU memory hovered around 15/32GB both before and after. FPS remained the same in both quality and performance modes respectively. Testing the exact same part of a game. Did both 1440p and AMD FSR scaling - no fluctuation in results both before and after.

Cheers for the effort and if it improves things for you, great! I'm just happy to be done with my fear that I've only been using 50% of my GPU for the past year lol

**Comment by IvanButBetter** [comment]
You can make sure the VM is 100% not using any resources by fully shutting it down after use (Not just disconnecting).

Open `Hyper-V Manager`
On the left-hand side, under the `Hyper-V Manager` dropdown you will see the name of your host machine. Click on it
In the centre tab you will see `Virtual Machines`, and the name of your VM. Click on it
Lastly, on the right hand side you will see a button that says `Turn Off`. This will shut down the VM completely

**Comment by mlbrnm** [comment]
Is there a way to change the GPU partition percentage after you've built the machine?

> I Frankenstein-ed this one together... Make sure you run as Administrator. [(De)attach.zip](https://github.com/user-attachments/files/17000225/De.attach.zip)

Thanks for making that.

Is this the best way to change the GPU allocation too? I set mine originally to 20% to play it safe, but I'm thinking I want higher now.
